homeassistant/components/hassio.py

```diff
@@ -0,0 +1,232 @@
+            yield from hassio.send_command(
+                "/supervisor/options", payload=service.data)
+        elif service.service == SERVICE_HOMEASSISTANT_UPDATE:
+            yield from hassio.send_command(
+                "/homeassistant/update", payload=service.data)
+
+    hass.services.async_register(
+        DOMAIN, SERVICE_HOST_SHUTDOWN,
+        descriptions[DOMAIN][SERVICE_HOST_SHUTDOWN])
+    hass.services.async_register(
+        DOMAIN, SERVICE_HOST_REBOOT,
+        descriptions[DOMAIN][SERVICE_HOST_REBOOT])
+    hass.services.async_register(
+        DOMAIN, SERVICE_HOST_REBOOT, descriptions[DOMAIN][SERVICE_HOST_REBOOT],
+        schema=SCHEMA_SERVICE_UPDATE)
+    hass.services.async_register(
+        DOMAIN, SERVICE_SUPERVISOR_UPDATE,
+        descriptions[DOMAIN][SERVICE_SUPERVISOR_UPDATE],
+        schema=SCHEMA_SERVICE_UPDATE)
+    hass.services.async_register(
+        DOMAIN, SERVICE_SUPERVISOR_OPTIONS,
+        descriptions[DOMAIN][SERVICE_SUPERVISOR_OPTIONS],
+        schema=SCHEMA_SERVICE_SUPERVISOR_OPTIONS)
+    hass.services.async_register(
+        DOMAIN, SERVICE_NETWORK_OPTIONS,
+        descriptions[DOMAIN][SERVICE_NETWORK_OPTIONS],
+        schema=SCHEMA_SERVICE_NETWORK_OPTIONS)
+
+    @asyncio.coroutine
+    def async_service_handler_addons(service):
+        """Handle HassIO service calls for addons."""
+        addon = service.data[ATTR_ADDON]
+        options = service.data.get(ATTR_OPTIONS)
+
+        # extract version
+        if ATTR_VERSION in service.data:
+            version = {ATTR_VERSION: service.data[ATTR_VERSION]}
+        else:
+            version = None
+
+        if service.service == SERVICE_ADDON_INSTALL:
+            yield from hassio.send_command(
+                "/addons/{}/install".format(addon), payload=version)
+        elif service.service == SERVICE_ADDON_UNINSTALL:
+            yield from hassio.send_command(
+                "/addons/{}/uninstall".format(addon))
+        elif service.service == SERVICE_ADDON_START:
+            yield from hassio.send_command(
+                "/addons/{}/start".format(addon), payload=options)
+        elif service.service == SERVICE_ADDON_STOP:
+            yield from hassio.send_command("/addons/{}/stop".format(addon))
+        elif service.service == SERVICE_ADDON_UPDATE:
+            yield from hassio.send_command(
+                "/addons/{}/update".format(addon), payload=version)
+
+    hass.services.async_register(
+        DOMAIN, SERVICE_ADDON_UPDATE,
+        descriptions[DOMAIN][SERVICE_ADDON_UPDATE],
+        schema=SCHEMA_SERVICE_ADDONS_VERSION)
+    hass.services.async_register(
+        DOMAIN, SERVICE_ADDON_INSTALL,
+        descriptions[DOMAIN][SERVICE_ADDON_INSTALL],
+        schema=SCHEMA_SERVICE_ADDONS_VERSION)
+    hass.services.async_register(
+        DOMAIN, SERVICE_ADDON_UNINSTALL,
+        descriptions[DOMAIN][SERVICE_ADDON_UNINSTALL],
+        schema=SCHEMA_SERVICE_ADDONS)
+    hass.services.async_register(
```

Comment:
We have no access from outside to supervisor. I think that is the best thing for security. So we need passtrough this stuff over hass.

I design the thing in the way that we call /xy/info with config pannel to load data and send it to frontend. So we need only call this getter (like the code from hassbian config) to rebuild infos.